src/middlewared/middlewared/plugins/jail.py

```diff
@@ -1240,10 +1240,19 @@ def progress_callback(content, exception):
 
         return True
 
-    @accepts(Str("jail"))
-    @job(lock=lambda args: f"jail_export:{args[-1]}")
-    def export(self, job, jail):
-        """Exports jail to zip file"""
+    @accepts(
+        Dict(
+            'options',
+            Str('jail', required=True),
+            Str('compression_algorithm', default='ZIP', enum=['ZIP', 'LZMA'])
+        )
+    )
+    @job(lock=lambda args: f'jail_export:{args[-1].get("jail")}')
+    def export(self, job, options):
+        """
+        Exports jail to compressed file.
```

Comment:
I would use present tense for docstrings.  It's shorter and often clearer.
```suggestion
        Export jail to compressed file.
```